lib/api/core/validation/types/geoShape.js

```diff
@@ -1,196 +1,223 @@
+    let
+      coordinateValidation,
+      result = true;
+
+    if (!this.checkStructure(allowedShapes, shape, errorMessages)) {
+      return false;
+    }
+
+    switch (shape.type) {
+      case 'point':
+      case 'multipoint':
+        coordinateValidation = isPoint;
+        break;
+      case 'linestring':
+      case 'multilinestring':
+        coordinateValidation = isLine;
+        break;
+      case 'polygon':
+      case 'multipolygon':
+        coordinateValidation = isPolygon;
+        if (shape.orientation && allowedOrientations.indexOf(shape.orientation) === -1) {
+          errorMessages.push('The orientation property has not a valid value.');
           result = false;
         }
-      }
-      break;
-    case 'envelope':
-      coordinateValidation = isEnvelope;
-      break;
-    case 'circle':
-      coordinateValidation = isPoint;
-      if (typeof shape.radius === 'string') {
-        try {
-          if (typeof convertDistance(shape.radius) !== 'number') {
+        break;
+      case 'geometrycollection':
+        coordinateValidation = () => true;
+
+        for (let i = 0; i < shape.geometries.length; i++) {
+          if (!this.recursiveShapeValidation(allowedShapes, shape.geometries[i], errorMessages)) {
+            result = false;
+          }
+        }
+        break;
+      case 'envelope':
+        coordinateValidation = isEnvelope;
+        break;
+      case 'circle':
+        coordinateValidation = isPoint;
+        if (typeof shape.radius !== 'number') {
+          return true;
+        }
+
+        if (typeof shape.radius === 'string') {
+          try {
+            if (typeof convertDistance(shape.radius) !== 'number') {
+              errorMessages.push('The radius property has not a valid format.');
+              result = false;
+            }
+          }
+          catch (error) {
             errorMessages.push('The radius property has not a valid format.');
             result = false;
           }
         }
-        catch (error) {
+        else {
           errorMessages.push('The radius property has not a valid format.');
           result = false;
         }
-      }
-      else if (typeof shape.radius !== 'number') {
-        errorMessages.push('The radius property has not a valid format.');
+        break;
+      default:
+        errorMessages.push(`Unrecognized shape type "${shape.type}"`);
         result = false;
-      }
-      break;
-  }
+    }
 
-  if (isMulti) {
-    if (shape.coordinates.some(coordinate => !coordinateValidation(coordinate))) {
-      errorMessages.push(`One of the shapes in  the shape type "${shape.type}" has bad coordinates.`);
-      result = false;
+    if (isMulti) {
+      if (shape.coordinates.some(coordinate => !coordinateValidation(coordinate))) {
+        errorMessages.push(`One of the shapes in  the shape type "${shape.type}" has bad coordinates.`);
+        result = false;
+      }
+    }
+    else if (!coordinateValidation(shape.coordinates)) {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL') Add the missing "else" clause. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AElseIfWithoutElse)